src/main/kotlin/cacophony/controlflow/functions/GenerateCall.kt

```diff
@@ -66,10 +89,41 @@ fun generateCall(
     }
 
     nodes.add(CFGNode.Call(function))
-    nodes.add(CFGNode.AdditionAssignment(rsp, CFGNode.ConstantLazy { alignmentShift.value + 8 * stackArguments.size }))
 
-    if (result != null)
-        nodes.add(CFGNode.Assignment(CFGNode.RegisterUse(result), CFGNode.RegisterUse(Register.FixedRegister(HardwareRegister.RAX))))
+    if (result == null) {
+        nodes.add(
+            CFGNode.AdditionAssignment(
+                rsp,
+                CFGNode.ConstantLazy {
+                    alignmentShift.value + 8 * stackArguments.size +
+                        8 * stackResultsSize
+                },
+            ),
+        )
+    } else {
+        if (stackResultsSize == 0) {
```

Comment:
I think we don't need that much ifology, we can just do `Addition(rsp, 8*stackArguments.size)->read results->Addition(rsp, alignmentShift)` - logic should remain the same

Reply:
This would always result in two additions even if they are not necessary, and also break tests :weary: